tools/flowc/backends/nim/runtime.nim

```diff
@@ -8,68 +8,341 @@ import hashes
 import asyncdispatch
 import osproc
 import macros
+import system/widestrs
 import "flow_lib/httpServer_type"
 
 # Runtime for NIM backend
 
 {.experimental: "overloadableEnums".}
 
-proc rt_runtime_error*(what: string): void =
-  assert(false, "runtime error: " & what)
-
-proc rt_escape(s: string): string = 
-  var r: string = ""
-  for ch in s:
-    case ch:
-    of '\n': r.add("\\n")
-    of '\t': r.add("\\t")
-    of '\r': r.add("\\u000d")
-    of '\\': r.add("\\\\")
-    of '"':  r.add("\\\"")
-    else: r.add(ch)
-  return r
+#[ Flow string: either a UTF-16 string of UTF-8 ]#
+
+const use16BitString = when defined UTF8: false else: true
+
+when use16BitString:
+  type String = seq[Utf16Char]
+else:
+  type String = string
```

Comment:
Just uppercase String is a quite generic name. Maybe "RtString" would be a better name.

Reply:
Ok, agree, String is renamed to RtString.